beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -214,6 +217,37 @@ func (s *Service) validateRangeRequest(r *pb.BeaconBlocksByRangeRequest) error {
 	return nil
 }
 
+// filters all the provided blocks to ensure they are canonical
+// and are strictly linear.
+func (s *Service) filterBlocks(ctx context.Context, blks []*ethpb.SignedBeaconBlock, roots [][32]byte, prevRoot *[32]byte,
```

Comment:
what does `prevRoot` mean? is it like a starting head root?

Reply:
I added a comment in the main request handler of what prevRoot is and why it is used.